src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml

```diff
@@ -21,5 +21,11 @@
     <type fullname="System.StartupHookProvider" feature="System.StartupHookProvider.IsSupported" featurevalue="false">
       <method signature="System.Boolean get_IsSupported()" body="stub" value="false" />
     </type>
+    <type fullname="System.Resources.ManifestBasedResourceGroveler" feature="System.Resources.AllowReflectionForNonPrimitiveTypes" featurevalue="false">
```

Comment:
Can we combine both the code and these entries into a single place that holds this bool? I think it makes more sense to have a single property for this - (maybe a static on `ResourceManager`?).

I also kind of think we should have `ResourceManager` in the name of the switch. What do you think about `System.Resources.ResourceManager.CustomResourceTypes.IsSupported`? To follow with the current naming pattern. I'm open to other ideas. I kind of don't like putting `Reflection` in the name, as that is too much of an implementation detail IMO.

We should update https://github.com/dotnet/runtime/blob/master/docs/workflow/trimming/feature-switches.md with this new switch.